v2/internal/controllers/reconcile_policy_test.go

```diff
@@ -77,12 +78,63 @@ func Test_ReconcilePolicy_UnknownPolicyIsIgnored(t *testing.T) {
 
 func Test_ReconcilePolicy_DetachOnDelete_SkipsDelete(t *testing.T) {
 	t.Parallel()
-	testDeleteSkipped(t, "detach-on-delete")
+	testDeleteSkipped(t, "detach-on-delete", false)
 }
 
 func Test_ReconcilePolicy_Skip_SkipsDelete(t *testing.T) {
 	t.Parallel()
-	testDeleteSkipped(t, "skip")
+	testDeleteSkipped(t, "skip", false)
+}
+
+func Test_ReconcilePolicyOnNamespace_DetachOnDelete_SkipsDelete(t *testing.T) {
+	t.Parallel()
+	testDeleteSkipped(t, "detach-on-delete", true)
+}
+
+func Test_ReconcilePolicyOnNamespace_Skip_SkipsDelete(t *testing.T) {
+	t.Parallel()
+	testDeleteSkipped(t, "skip", true)
+}
```

Comment:
Should these three (four?) be a table test?

Reply:
Changed to a table test.